pkg/smokescreen/conntrack/instrumented_conn_test.go

```diff
@@ -27,26 +27,33 @@ func TestInstrumentedConnByteCounting(t *testing.T) {
 	tr := NewTestTracker(0)
 	sent := []byte("X-Smokescreen-Test")
 
-	var wg sync.WaitGroup
-	wg.Add(2)
-
+	writerErrChan := make(chan error, 1)
 	go func() {
-		defer wg.Done()
+		defer close(writerErrChan)
```

Comment:
This is a much better pattern which makes it easier to pass back contextual errors.